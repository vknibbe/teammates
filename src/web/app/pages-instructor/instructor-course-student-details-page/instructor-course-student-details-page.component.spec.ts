import { HttpClientTestingModule } from '@angular/common/http/testing';
import { Component, Input } from '@angular/core';
import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { MatSnackBarModule } from '@angular/material';
import { RouterTestingModule } from '@angular/router/testing';
import { JoinState, Student } from '../../../types/api-output';
import { Gender } from '../../../types/gender';
import { StudentProfile } from '../student-profile/student-profile';
import {
  InstructorCourseStudentDetailsPageComponent,
} from './instructor-course-student-details-page.component';

@Component({ selector: 'tm-student-profile', template: '' })
class StudentProfileStubComponent {
  @Input() studentProfile: StudentProfile | undefined;
  @Input() studentName: string = '';
  @Input() photoUrl: string = '/assets/images/profile_picture_default.png';
  @Input() hideMoreInfo: boolean = false;
}
@Component({ selector: 'tm-course-related-info', template: '' })
class CourseRelatedInfoStubComponent {
  @Input() student: Student = {
    email: '',
    courseId: '',
    name: '',
    lastName: '',
    comments: '',
    teamName: '',
    sectionName: '',
    joinState: JoinState.JOINED,
  };
}
@Component({ selector: 'tm-more-info', template: '' })
class MoreInfoStubComponent {
  @Input() studentName: string = '';
  @Input() moreInfoText: string = '';
}

describe('InstructorCourseStudentDetailsPageComponent', () => {
  let component: InstructorCourseStudentDetailsPageComponent;
  let fixture: ComponentFixture<InstructorCourseStudentDetailsPageComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        InstructorCourseStudentDetailsPageComponent,
        StudentProfileStubComponent,
        CourseRelatedInfoStubComponent,
        MoreInfoStubComponent,
      ],
      imports: [
        HttpClientTestingModule,
        RouterTestingModule,
        MatSnackBarModule,
      ],
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(InstructorCourseStudentDetailsPageComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should snap with default fields', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('should snap with populated student', () => {
    component.student = {
      email: 'studentEmail@email.com',
      courseId: 'CS3281',
      name: 'firstName',
      lastName: 'lastName',
      comments: 'This is a comment',
      teamName: 'myTeam',
      sectionName: 'mySection',
      joinState: JoinState.JOINED,
    };
    component.studentProfile = {
      name: 'name',
      shortName: 'shortName',
      email: 'profileEmail@email.com',
      institute: 'NUS',
      nationality: 'Indian',
      gender: Gender.MALE,
      moreInfo: 'I have more info here',
    };
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });
});
