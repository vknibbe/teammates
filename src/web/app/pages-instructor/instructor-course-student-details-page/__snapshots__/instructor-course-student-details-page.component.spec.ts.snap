// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstructorCourseStudentDetailsPageComponent should snap with default fields 1`] = `
<tm-instructor-course-student-details-page
  photoUrl={[Function String]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
>
  
</tm-instructor-course-student-details-page>
`;

exports[`InstructorCourseStudentDetailsPageComponent should snap with populated student 1`] = `
<tm-instructor-course-student-details-page
  photoUrl={[Function String]}
  route={[Function ActivatedRoute]}
  statusMessageService={[Function StatusMessageService]}
  student={[Function Object]}
  studentProfile={[Function Object]}
  studentProfileService={[Function StudentProfileService]}
  studentService={[Function StudentService]}
>
  <h1>
    firstName
  </h1><tm-student-profile
    ng-reflect-photo-url="http://localhost:8080/webapi/s"
    ng-reflect-student-profile="[object Object]"
  /><tm-course-related-info
    ng-reflect-student="[object Object]"
  /><tm-more-info
    ng-reflect-more-info-text="I have more info here"
    ng-reflect-student-name="firstName"
  />
</tm-instructor-course-student-details-page>
`;
